connect/cli/plugins/project/helpers.py

```diff
@@ -0,0 +1,151 @@
+#  Copyright © 2021 CloudBlue. All rights reserved.
+
+from cookiecutter.main import cookiecutter
+from cookiecutter.config import DEFAULT_CONFIG
+from cookiecutter.utils import rmtree
+from cookiecutter.exceptions import OutputDirExistsException
+
+from connect.cli.plugins.project.constants import (
+    PROJECT_BOILERPLATE_URL,
+)
+
+import json
+import os
+import sys
+import inspect
+import importlib
+import tempfile
+import shutil
+
+import click
+from click import ClickException
+
+from connect.reports.validator import (
+    validate,
+    validate_with_schema,
+)
+from connect.reports.parser import parse
+
+
+def bootstrap_project(data_dir: str):
+    click.secho('Bootstraping report project...\n', fg='blue')
+
+    _purge_cookiecutters_dir()
+
+    try:
+        project_dir = cookiecutter(PROJECT_BOILERPLATE_URL, output_dir=data_dir)
+        click.secho(f'\nReport Project location: {project_dir}', fg='blue')
+    except OutputDirExistsException as error:
+        project_path = str(error).split('"')[1]
+        raise ClickException(
+            f'\nThe directory "{project_path}" is already created, '
+            '\nif you would like to use that name, please delete '
+            'the directory or use another location.',
+        )
+
+
+def validate_project(project_dir):
+    click.secho(f'Validating project {project_dir}...\n', fg='blue')
+
+    data = _file_descriptor_validations(project_dir)
+
+    errors = validate_with_schema(data)
+    if errors:
+        raise ClickException(f'Invalid `reports.json`: {errors}')
+
+    report_project = parse(project_dir, data)
+
+    errors = validate(report_project)
+    if errors:
+        raise ClickException(f'Invalid `reports.json`: {",".join(errors)}')
+
+    for report in report_project.reports:
+        _entrypoint_validations(project_dir, report.entrypoint, report.report_spec)
+
+    click.secho(f'Report Project {project_dir} has been successfully validated.', fg='green')
+
+
+def add_report(project_dir, package_name):
+    if not os.path.isdir(f'{project_dir}/{package_name}'):
+        raise ClickException(
+            f'The directory package called `{package_name}` does not exist,'
+            '\nPlease, create it or choose an existing one using `-n` option.',
+        )
+
+    click.secho(f'Adding new report to project {project_dir}...\n', fg='blue')
+
+    _purge_cookiecutters_dir()
+
+    with tempfile.TemporaryDirectory() as add_report_tmpdir:
+        report_name = input(
+            'report_name [My New Report]: ',
+        ) or 'My New Report'
+        report_description = input(
+            'report_description [This report provides all data I need]: ',
+        ) or 'This report provides all data I need'
+        report_author = input(
+            'author [Globex Corporation]: ',
+        ) or 'Globex Corporation'
+
+        report_dir = cookiecutter(
```

Comment:
Even better to read the cookiecutter.json file and prompt the user for each property that does not belong to the list of property related to a repository. This way, if we add a new property for reports to the template (like renderers for example) it will still work.

Reply:
done!